lib/spack/spack/util/compression.py

```diff
@@ -77,8 +77,12 @@ def _system_untar(archive_file):
         archive_file (str): absolute path to the archive to be extracted.
         Can be one of .tar(.[gz|bz2|xz|Z]) or .(tgz|tbz|tbz2|txz).
     """
-    outfile = os.path.basename(strip_extension(archive_file, "tar"))
-
+    archive_file_no_ext = strip_extension(archive_file, "tar")
+    outfile = os.path.basename(archive_file_no_ext)
+    if archive_file_no_ext == archive_file:
+        # the archive file has no extension. Tar on windows cannot untar onto itself
+        archive_file = archive_file_no_ext + ".tar"
+        shutil.copy(archive_file_no_ext, archive_file)
```

Comment:
> I think this should be a move vs. a copy (I assume there's no issue in the sense that the untar overwrites the copy with the original name, but my issue is more with creating duplicates of the file when it isn't necessary).

Sounds good.

Reply:
> Also, I think it's worth choosing a modified name other than adding .tar, since the main significant point of this is to just choose a name different than the intended final unpacked directory. (something like archive_file = archive_file_no_ext + "-input").
There are two problems I've seen: 1. tar on windows complains it cannot untar a file onto itself. 2. on unix unzip complains when there is no extension. So '-input' won't work on issue 2. Adding the proper extension solves the issues consistently.